contracts/exchange/test/transactions_unit_tests.ts

```diff
@@ -0,0 +1,707 @@
+            const transaction = await generateZeroExTransactionAsync();
+            const transactionHash = transactionHashUtils.getTransactionHashHex(transaction);
+            const actualGasPrice = transaction.gasPrice.plus(1);
+            const expectedError = new ExchangeRevertErrors.TransactionGasPriceError(
+                transactionHash,
+                actualGasPrice,
+                transaction.gasPrice,
+            );
+            expect(
+                transactionsContract.assertExecutableTransaction.callAsync(transaction, randomSignature(), {
+                    gasPrice: actualGasPrice,
+                }),
+            ).to.revertWith(expectedError);
+        });
+        it('should revert if currentContextAddress is non-zero', async () => {
+            await transactionsContract.setCurrentContextAddress.awaitTransactionSuccessAsync(accounts[0]);
+            const transaction = await generateZeroExTransactionAsync();
+            const transactionHash = transactionHashUtils.getTransactionHashHex(transaction);
+            const expectedError = new ExchangeRevertErrors.TransactionInvalidContextError(transactionHash, accounts[0]);
+            expect(
+                transactionsContract.assertExecutableTransaction.callAsync(transaction, randomSignature()),
+            ).to.revertWith(expectedError);
+        });
+        it('should revert if the transaction has already been executed', async () => {
+            const transaction = await generateZeroExTransactionAsync();
+            const transactionHash = transactionHashUtils.getTransactionHashHex(transaction);
+            await transactionsContract.setTransactionExecuted.awaitTransactionSuccessAsync(transactionHash);
+            const expectedError = new ExchangeRevertErrors.TransactionError(
+                ExchangeRevertErrors.TransactionErrorCode.AlreadyExecuted,
+                transactionHash,
+            );
+            expect(
+                transactionsContract.assertExecutableTransaction.callAsync(transaction, randomSignature()),
+            ).to.revertWith(expectedError);
+        });
+        it('should revert if signer != msg.sender and the signature is invalid', async () => {
+            const transaction = await generateZeroExTransactionAsync({ signerAddress: accounts[0] });
+            const transactionHash = transactionHashUtils.getTransactionHashHex(transaction);
+            const expectedError = new ExchangeRevertErrors.TransactionSignatureError(
+                transactionHash,
+                accounts[0],
+                INVALID_SIGNATURE,
+            );
+            expect(
+                transactionsContract.assertExecutableTransaction.callAsync(transaction, INVALID_SIGNATURE, {
+                    from: accounts[1],
+                }),
+            ).to.revertWith(expectedError);
+        });
+        it('should be successful if signer == msg.sender and the signature is invalid', async () => {
```

Comment:
I think that we should also check that the `tx` should be successful if `signer == msg.sender and the signature is valid`. It's a bit pedantic, but it would make the suite more complete.